src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1805,14 +1805,20 @@ boolean getAndClearInterrupt() {
         if (com.ibm.oti.vm.VM.isJVMInSingleThreadedMode()) {
             return interruptedImpl();
         }
-        boolean oldValue = interrupted;
+
+        /*
+         * It is possible for the Java and native interrupt status to get out of sync. Sending
+         * two interrupts to the same thread in quick succession can cause this. It is critical
+         * that clearing the interrupt status also clears the native status. If the Java
+         * interrupt status is false, it is important to verify that this is consistent with the
+         * native status to prevent an unexpected interrupt exception the next time
+         * sleep() or wait() is called.
+         */
+        boolean oldValue = interrupted || isInterruptedImpl();
         if (oldValue) {
             synchronized (interruptLock) {
-                oldValue = interrupted;
-                if (oldValue) {
-                    interrupted = false;
-                    clearInterruptEvent();
-                }
+                interrupted = false;
```

Comment:
Please include the comment present in jdk24 and the head stream:
```java
                // We may have been interrupted the moment after we read the field,
                // so only clear the field if we saw that it was set and will return
                // true; otherwise we could lose an interrupt.
```